yaqd_andor/_andor_sdk2.py

```diff
@@ -0,0 +1,248 @@
+            "DRV_P11INVALID": 20087,
+            "DRV_USBERROR": 20089,
+            "DRV_IOCERROR": 20090,
+            "DRV_VRMVERSIONERROR": 20091,
+            "DRV_GATESTEPERROR": 20092,
+            "DRV_USB_INTERRUPT_ENDPOINT_ERROR": 20093,
+            "DRV_RANDOM_TRACK_ERROR": 20094,
+            "DRV_INVALID_TRIGGER_MODE": 20095,
+            "DRV_LOAD_FIRMWARE_ERROR": 20096,
+            "DRV_DIVIDE_BY_ZERO_ERROR": 20097,
+            "DRV_INVALID_RINGEXPOSURES": 20098,
+            "DRV_BINNING_ERROR": 20099,
+            "DRV_INVALID_AMPLIFIER": 20100,
+            "DRV_INVALID_COUNTCONVERT_MODE": 20101,
+            "DRV_USB_INTERRUPT_ENDPOINT_TIMEOUT": 20102,
+            "DRV_ERROR_NOCAMERA": 20990,
+            "DRV_NOT_SUPPORTED": 20991,
+            "DRV_NOT_AVAILABLE": 20992,
+            "DRV_ERROR_MAP": 20115,
+            "DRV_ERROR_UNMAP": 20116,
+            "DRV_ERROR_MDL": 20117,
+            "DRV_ERROR_UNMDL": 20118,
+            "DRV_ERROR_BUFFSIZE": 20119,
+            "DRV_ERROR_NOHANDLE": 20121,
+            "DRV_GATING_NOT_AVAILABLE": 20130,
+            "DRV_FPGA_VOLTAGE_ERROR": 20131,
+            "DRV_OW_CMD_FAIL": 20150,
+            "DRV_OWMEMORY_BAD_ADDR": 20151,
+            "DRV_OWCMD_NOT_AVAILABLE": 20152,
+            "DRV_OW_NO_SLAVES": 20153,
+            "DRV_OW_NOT_INITIALIZED": 20154,
+            "DRV_OW_ERROR_SLAVE_NUM": 20155,
+            "DRV_MSTIMINGS_ERROR": 20156,
+            "DRV_OA_NULL_ERROR": 20173,
+            "DRV_OA_PARSE_DTD_ERROR": 20174,
+            "DRV_OA_DTD_VALIDATE_ERROR": 20175,
+            "DRV_OA_FILE_ACCESS_ERROR": 20176,
+            "DRV_OA_FILE_DOES_NOT_EXIST": 20177,
+            "DRV_OA_XML_INVALID_OR_NOT_FOUND_ERROR": 20178,
+            "DRV_OA_PRESET_FILE_NOT_LOADED": 20179,
+            "DRV_OA_USER_FILE_NOT_LOADED": 20180,
+            "DRV_OA_PRESET_AND_USER_FILE_NOT_LOADED": 20181,
+            "DRV_OA_INVALID_FILE": 20182,
+            "DRV_OA_FILE_HAS_BEEN_MODIFIED": 20183,
+            "DRV_OA_BUFFER_FULL": 20184,
+            "DRV_OA_INVALID_STRING_LENGTH": 20185,
+            "DRV_OA_INVALID_CHARS_IN_NAME": 20186,
+            "DRV_OA_INVALID_NAMING": 20187,
+            "DRV_OA_GET_CAMERA_ERROR": 20188,
+            "DRV_OA_MODE_ALREADY_EXISTS": 20189,
+            "DRV_OA_STRINGS_NOT_EQUAL": 20190,
+            "DRV_OA_NO_USER_DATA": 20191,
+            "DRV_OA_VALUE_NOT_SUPPORTED": 20192,
+            "DRV_OA_MODE_DOES_NOT_EXIST": 20193,
+            "DRV_OA_CAMERA_NOT_SUPPORTED": 20194,
+            "DRV_OA_FAILED_TO_GET_MODE": 20195,
+            "DRV_OA_CAMERA_NOT_AVAILABLE": 20196,
+            "DRV_PROCESSING_FAILED": 20211,
+        }
+
+        keyout = []
+        for key, val in errordict.items():
+            if val == code:
+                keyout.append(key)
+        if len(keyout) > 0:
+            return str(keyout[0])
+        else:
+            return ValueError("Code not found in error dictionary")
```

Comment:
```suggestion
        code_to_str = {v: k for k, v in errordict.items()}
        return code_to_str[val]
```
less confusing to just work with the dictionary you want (instead of the inverse).  

Also, no need to add an error, it will naturally be raised when the key does not exist.

Reply:
This entire function is just a dictonary lookup--I would suggest moving this dictionary outside of the scope of the function to be an attribute of the class.  Then lookups are more direct e.g. `self.code_to_str[code]`.

Reply:
got it